src/components/Games/tetris.js

```diff
@@ -0,0 +1,297 @@
+          if (playfield[row][col]) {
+            const name = playfield[row][col];
+            context.fillStyle = colors[name];
+            // Drawing 1 px smaller than the grid creates a grid effect
+            context.fillRect(col * grid, row * grid, grid-1, grid-1);
+          }
+        }
+      }
+
+      // Draw the active tetromino
+      if (tetromino) {
+        // Tetromino falls every 35 frames
+        if (++count > 35) {
+          setTetromino(prevTetromino => {
+            const newRow = prevTetromino.row + 1;
+            if (!isValidMove(prevTetromino.matrix, newRow, prevTetromino.col)) {
+              placeTetromino();
+              return prevTetromino;
+            }
+            return { ...prevTetromino, row: newRow };
+          });
+          setCount(0);
+        }
+
+        context.fillStyle = colors[tetromino.name];
+
+        for (let row = 0; row < tetromino.matrix.length; row++) {
+          for (let col = 0; col < tetromino.matrix[row].length; col++) {
+            if (tetromino.matrix[row][col]) {
+              // Drawing 1 px smaller than the grid creates a grid effect
+              context.fillRect((tetromino.col + col) * grid, (tetromino.row + row) * grid, grid-1, grid-1);
+            }
+          }
+        }
+      }
+    };
+
+    // Start the game loop
+    rAF = requestAnimationFrame(loop);
+
+    // Cleanup function
+    return () => cancelAnimationFrame(rAF);
+  }, [playfield, tetromino, tetrominoSequence, count]);
+
+  // Listen to keyboard events to move the active tetromino
+  const handleKeyDown = (e) => {
+    if (gameOver) return;
+
+    // Left and right arrow keys (move)
+    if (e.which === 37 || e.which === 39) {
+      setTetromino(prevTetromino => {
+        const newCol = e.which === 37 ? prevTetromino.col - 1 : prevTetromino.col + 1;
+        if (isValidMove(prevTetromino.matrix, prevTetromino.row, newCol)) {
+          return { ...prevTetromino, col: newCol };
+        }
+        return prevTetromino;
+      });
+    }
+
+    // Up arrow key (rotate)
+    if (e.which === 38) {
+      setTetromino(prevTetromino => {
+        const newMatrix = rotate(prevTetromino.matrix);
+        if (isValidMove(newMatrix, prevTetromino.row, prevTetromino.col)) {
+          return { ...prevTetromino, matrix: newMatrix };
+        }
+        return prevTetromino;
+      });
+    }
+
+    // Down arrow key (drop)
+    if (e.which === 40) {
+      setTetromino(prevTetromino => {
+        const newRow = prevTetromino.row + 1;
+        if (!isValidMove(prevTetromino.matrix, newRow, prevTetromino.col)) {
+          placeTetromino();
+          return prevTetromino;
+        }
+        return { ...prevTetromino, row: newRow };
+      });
+    }
+  };
+
+  return (
+    <canvas
+      id="game"
+      width="320"
+      height="640"
+      onKeyDown={handleKeyDown}
+      tabIndex="0"
+      style={{ border: '1px solid white' }}
+    />
+  );
+};
+
+export default TetrisGame;
```

Comment:
Please let me know how can I dispose of the game score and start game
need these for interface latter one